File: kbmf1k1k/kbmf1k1k_supervised_classification_variational_test.m
% Mehmet Gonen (mehmet.gonen@gmail.com)

function prediction = kbmf1k1k_supervised_classification_variational_test(Kx, Kz, state)
    prediction.Gx.mu = state.Ax.mu' * Kx;

    prediction.Gz.mu = state.Az.mu' * Kz;

    prediction.F.mu = prediction.Gx.mu' * prediction.Gz.mu;
end

File: kbmf1k1k/kbmf1k1k_supervised_classification_variational_test.R
# Mehmet Gonen (mehmet.gonen@gmail.com)

kbmf1k1k_supervised_classification_variational_test <- function(Kx, Kz, state) {
  Gx <- list(mean = crossprod(state$Ax$mean, Kx))

  Gz <- list(mean = crossprod(state$Az$mean, Kz))

  F <- list(mean = crossprod(Gx$mean, Gz$mean))

  prediction <- list(Gx = Gx, Gz = Gz, F = F)
}